include/tscpp/util/Histogram.h

```diff
@@ -0,0 +1,188 @@
+/** @file
+ *
+ * Fast small foot print histogram support.
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#pragma once
+
+#include <stdint.h>
+#include <array>
+
+#if !defined(TS_VERSION_NS)
+#define TS_VERSION_NS _10_0_0
+#endif
+
+namespace ts
+{
+inline namespace TS_VERSION_NS
```

Comment:
Why do we need versioning for this but not others?

Reply:
We want it for others, but I didn't want this PR to be a massive one. You can see all of the libswoc code is versioned that way, so as it replaces other core libraries those will gain this protection as well.

Reply:
I don't care what external libraries do (I know a snapshot of libswoc is in our source tree but that doesn't matter). I feel like something similar was proposed before and it got some push back. If you are planning to have the versioning mechanism for others as well, you should probably propose it separately not only for keeping this PR small but also for a clear consensus. I don't see any necessity to have it in this PR.